Objective-C/Tests/DatabaseTest.m

```diff
@@ -582,6 +582,60 @@ - (void) testSaveDocWithDeletedConflictUsingConcurrencyControl: (int)concurrency
     [self cleanDB];
 }
 
+#pragma mark Save Conflict Handler
+
+- (void) testConflictHandler {
+    NSString* docID = @"doc1";
+    CBLMutableDocument* doc = [[CBLMutableDocument alloc] initWithID: docID];
+    [doc setString: @"Tiger" forKey: @"firstName"];
+    [self saveDocument: doc];
+    AssertEqual([self.db documentWithID: docID].generation, 1u);
+    
+    CBLMutableDocument* doc1a = [[self.db documentWithID: docID] toMutable];
+    CBLMutableDocument* doc1b = [[self.db documentWithID: docID] toMutable];
+    
+    [doc1a setString: @"Scotty" forKey: @"nickName"];
+    [self saveDocument: doc1a];
+    AssertEqual([self.db documentWithID: docID].generation, 2u);
+    
+    NSError* error;
+    [doc1b setString: @"Scott" forKey: @"nickName"];
+    Assert([self.db saveDocument: doc1b
+                 conflictHandler:^BOOL(CBLMutableDocument * document, CBLDocument * old) {
+                     AssertEqualObjects(doc1b, document);
```

Comment:
I think we can just do Assert(doc1b == document). I understand that using AssertEqualObjects will call equals method?

Reply:
Yes, true. Agree.